### PR TITLE
automl best practices

### PR DESCRIPTION
**Pull request by meganjkurka**
Hi everyone,

Can you let me know if I am doing the target encoding-esque step correctly? This is the last step.  I don't know if it will cause overfitting since cross validation is also being used inside AutoML.

### CODE REVIEW

File: best-practices/automl/README.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Best Practices - AutoML
```
Comment (ledell) [inline best-practices/automl/README.md:1]:
```suggestion
# Best Practices - H2O AutoML
```

Comment (ledell) [reply best-practices/automl/README.md:1]: Edit: To be clear from the beginning that we're specifically talking about H2O AutoML.

File: best-practices/automl/automl_best_practices.ipynb [hunk 2]
```diff
@@ -0,0 +1,1617 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Best Practices - AutoML\n",
```
Comment (ledell) [inline best-practices/automl/automl_best_practices.ipynb:7]:
```suggestion
    "# Best Practices - H2O AutoML\n",
```

File: best-practices/automl/automl_best_practices.ipynb [hunk 3]
```diff
@@ -0,0 +1,1617 @@
...
+    "* feature engineering"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Section 1 - Choosing the Correct Validation Data\n",
+    "\n",
+    "For this tutorial, we want to use AutoML to build a model to predict the price of Airbnb's: <https://www.kaggle.com/dgomonov/new-york-city-airbnb-open-data>.\n",
```
Comment (ledell) [inline best-practices/automl/automl_best_practices.ipynb:21]:
```suggestion
    "For this tutorial, we want to use H2O AutoML to build a model to predict the price of Airbnb's: <https://www.kaggle.com/dgomonov/new-york-city-airbnb-open-data>.\n",
```

File: best-practices/automl/README.md [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+This tutorial focuses on two areas for the data scientist to ensure the best performance by AutoML: 
+
+* choosing the correct validation data
+* feature engineering
+
+**Goal:** learn about recommended practices for AutoML.
+
+## The Data
+
+For this tutorial, we want to use AutoML to build a model to predict the price of Airbnb's: <https://www.kaggle.com/dgomonov/new-york-city-airbnb-open-data>.
```
Comment (ledell) [inline best-practices/automl/README.md:12]:
```suggestion
For this tutorial, we want to use H2O AutoML to build a model to predict the price of Airbnb's: <https://www.kaggle.com/dgomonov/new-york-city-airbnb-open-data>.
```

**Comment by meganjkurka** [comment]
@ledell @sebhrusen I removed the Target Encoding section since it is not supported by automl yet and I created a jira to add this functionality: https://0xdata.atlassian.net/browse/PUBDEV-7488

**Pull request merged**